ci: add dependabot config for updating vendored packages

nixpanic opened the pull request:
Vendored dependencies need updating on regular basis. This is currently
done manually by developers, but it can be automated by Dependabot. By
dropping the dependabot.yml config file in the .github/ directory the
bot should get enabled.

See-also: https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/enabling-and-disabling-version-updates

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

nixpanic committed:
ci: add dependabot config for updating vendored packages

Vendored dependencies need updating on regular basis. This is currently
done manually by developers, but it can be automated by Dependabot. By
dropping the dependabot.yml config file in the .github/ directory the
bot should get enabled.

See-also: https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/enabling-and-disabling-version-updates
Signed-off-by: Niels de Vos <ndevos@redhat.com>

yati1998 reviewed (commented):
LGTM

nixpanic commented:
Marking this `DNM`, as I want to check the subject of PRs that dependabot creates.

nixpanic commented:
Added an additional config option to set the `rebase: ` prefix that we expect in subjects of commit messages. We might need a change to the DCO/commitlint configuration as @dependabot should not add a `Signed-off-by` line.

pkalever commented on .github/dependabot.yml:
```diff
@@ -0,0 +1,11 @@
+  - package-ecosystem: "gomod"
+    directory: "/"
+    schedule:
+      interval: "weekly"
```
Why not daily?

nixpanic replied:
there are lots of dependencies, not sure how how often updates are published... we might want to change this later, when we are confident it is useful

pkalever replied:
Sure, makesense. :+1:

humblec replied:
its good to weekly as it can pull the consolidated changes together for few at time.. daily looks like a an overkill :)

pkalever replied:
I just want to avoid the need to trigger this manually on a release day, if that makesense.

pkalever commented:
This is a cool addition :+1:

nixpanic committed:
build: get gomodules.xyz/jsonpatch/v2 from github

The `make containerized-test TARGET=mod-check` jobs fail because the web
service at gomodules.xyz is not behaving correctly at the moment:

    running: go mod verify
    go: sigs.k8s.io/controller-runtime@v0.9.2 requires
    	gomodules.xyz/jsonpatch/v2@v2.2.0: unrecognized import path "gomodules.xyz/jsonpatch/v2": reading https://gomodules.xyz/jsonpatch/v2?go-get=1: 404 Not Found

Signed-off-by: Niels de Vos <ndevos@redhat.com>

nixpanic commented:
Added a commit to use `gomodules.xyz/jsonpatch/v2@v2.2.0` from GitHub, the gomodules.xyz service doesn't do redirection correctly at the moment and that causes mod-check to fail.

nixpanic commented:
It seems to be a slow Monday for repositories of Go packages :sob: 

**mod-check** continues to fail, but with an other issue:
```
running: go mod verify
go: github.com/libopenstorage/secrets@v0.0.0-20210709082113-dde442ea20ec requires
	github.com/hashicorp/vault@v1.4.2 requires
	github.com/hashicorp/vault-plugin-auth-cf@v0.5.4 requires
	github.com/cloudfoundry-community/go-cfclient@v0.0.0-20190201205600-f136f9222381 requires
	code.cloudfoundry.org/gofileutils@v0.0.0-20170111115228-4d0c80011a0f: unrecognized import path "code.cloudfoundry.org/gofileutils": https fetch: Get "https://code.cloudfoundry.org/gofileutils?go-get=1": x509: certificate signed by unknown authority
```

nixpanic committed:
build: vendor code.cloudfoundry.org/gofileutils from GitHub

There is a problem accessing the code.cloudfoundry.org web service iver
TLS. It seems to redirect to GitHub, so use the package from there:

    running: go mod verify
    go: github.com/libopenstorage/secrets@v0.0.0-20210709082113-dde442ea20ec requires
    	github.com/hashicorp/vault@v1.4.2 requires
    	github.com/hashicorp/vault-plugin-auth-cf@v0.5.4 requires
    	github.com/cloudfoundry-community/go-cfclient@v0.0.0-20190201205600-f136f9222381 requires
    	code.cloudfoundry.org/gofileutils@v0.0.0-20170111115228-4d0c80011a0f: unrecognized import path "code.cloudfoundry.org/gofileutils": https fetch: Get "https://code.cloudfoundry.org/gofileutils?go-get=1": x509: certificate signed by unknown authority

Signed-off-by: Niels de Vos <ndevos@redhat.com>

Rakshith-R commented:
/retest all

Rakshith-R commented:
> /retest all

running ci tests since this pr also modifies `go.mod` now, just to make sure everything still works :wink: 
Otherwise lgtm
Thanks!

humblec commented:
/retest ci/centos/mini-e2e/k8s-1.19

Rakshith-R commented:
/retest ci/centos/mini-e2e/k8s-1.19

Rakshith-R commented:
> /retest ci/centos/mini-e2e/k8s-1.19

failed due to #2264

nixpanic commented:
> > /retest all
> 
> running ci tests since this pr also modifies `go.mod` now, just to make sure everything still works wink
> Otherwise lgtm
> Thanks!

I do not think this is needed, the contents of the .go files under `vendor/` have not been changed, even the hashes of the replacement locations are the same.

Rakshith-R reviewed (approved):
I missed the vendor folder part 😅🤦.
Thanks!
